src/app/brands/[brandId]/page.tsx

```diff
@@ -0,0 +1,84 @@
+import BrandPage from "@/components/BrandPage/BrandPage";
+import ProductGallery from "@/components/ProductGallery";
+import { getBrand } from "@/model/BrandModel";
+import { getFavorites } from "@/model/FavoriteProductsModel";
+import { getProducts } from "@/model/ProductModel";
+import { Product } from "@/services/ProductAPI";
+import getUserServerSide from "@/services/server/UserService";
+import { Metadata } from "next";
+import { notFound } from "next/navigation";
+
+export async function generateMetadata({
+  params,
+}: {
+  params: { brandId: string };
+}): Promise<Metadata> {
+  let brand = await getBrand(params.brandId);
+
+  if (brand === undefined) {
+    return {
+      title: "Not found!",
+    };
+  }
+  return {
+    title: brand.name,
+  };
+}
+
+export default async function Page({
+  params,
+}: {
+  params: { brandId: string };
+}) {
+  const user = await getUserServerSide();
+  let products: Product[] = [];
+  let brand = undefined;
+
+  try {
+    [products, brand] = await Promise.all([
+      getProducts({
+        brandId: params.brandId,
+      }),
+      getBrand(params.brandId),
+    ]);
+  } catch (err) {
+    console.error(err);
```

Comment:
If this fails we just `console.error`? Shouldn't we show some information to the user?